articles/101_build_tool.md

```diff
@@ -0,0 +1,212 @@
+It should work with ROS 1 packages as well as ROS 2 packages which provide the necessary information in their manifest files.
+It would be great if the tool could also be used to build other packages (e.g. Gazebo including its dependencies).
+
+### Out of Scope
+
+The build tool does not cover the steps necessary to fetch the sources of the to-be-built packages.
+This can be done with existing tools (e.g. [rosinstall_generator](http://wiki.ros.org/rosinstall_generator), [vcstool](https://github.com/dirk-thomas/vcstool)).
+
+The build tool also does not provide a mechanism to install any dependencies required to build the packages.
+In the ROS ecosystem [rosdep](http://wiki.ros.org/rosdep) can be used for this.
+
+The build tool also does not create binary packages (e.g. a Debian package).
+In the ROS ecosystem [bloom](http://wiki.ros.org/bloom) is used to generate the required metadata and then platform dependent tools like `dpkg-buildpackage` build binary packages.
+
+## Build Tool vs. Build System
+
+A build tool operates on a set of packages.
+It determines the dependency graph and invokes the package specific build tool of each package in topological order.
+The build tool itself should know as little as possible about the build system used for a specific package.
+Just enough in order to know how to setup the environment for it, invoke the build, and setup the environment to use the built package.
+
+The build system on the other hand operates on a single package.
+In the case a package uses e.g. `CMake` the responsibility of the tool comes down to invoke the common steps `cmake`, `make`, `make install` for this package.
+As another example for a package using `Autotools` the steps could look like `configure`, `make`, `make install`.
+
+### Environment Setup
+
+A very important part beside the actual build of a package is the environment setup.
+For a CMake package e.g. a dependency discovered using the CMake function `find_package` needs to be either in a common location or the location must be provided through the environment variable `CMAKE_PREFIX_PATH`.
+On the other hand after a package has been built and installed the environment might need to be extended to use the package.
+E.g. when a package installs a shared library in a non-default location the environment variable `LD_LIBRARY_PATH` (or `PATH` on Windows) needs to be extended to include the containing folder.
+
+The functionality to setup these environment variables can be provided by either the build tool or the build system.
+In the latter case the build tool only needs to know how the build system exposes the environment setup in order to reuse it.
+
+Considering the use case that a user might want to invoke the build system of each package manually it is beneficial if the build system already provides as much of the environment setup as possible.
+That avoids that the user has to manually take care of the environment setup when not using a build tool.
+
```

Comment:
This might be better worded as:

This avoids forcing the user to manually take care of the environment setup when not using a build tool.